### PR TITLE
feat: render design examples in daffio

### LINKED ISSUE #2903: [FEAT] Display Example Viewer in Markdown Files
# :bulb: Feature request

## Feature Name
Display Example Viewer in Markdown Files

## The Desired Behavior
- [x] Extract `DesignLandExampleViewerContainer` into docs-utils
- [x] Add support to daff.io for `DesignLandExampleViewerContainer`

## Your Use Case
As a designer and developer, I expect the example viewer in component markdown files to be visible.

As seen in the screen shots, the `<design-land-example-viewer-container>` is added to the readme for card. It's not visible on https://next.daff.io/docs/design/components/card, but it's shown in the dom when you inspect the page.

Screen Shot 2024-07-08 at 2 16 22 PM
Screen Shot 2024-07-08 at 2 16 41 PM
Screen Shot 2024-07-08 at 2 18 08 PM

## Environment

<pre><code>
Daffodil version: X.Y.Z
Angular version: X.Y.Z 

 
For Tooling issues:
- Node version: XX  
- Platform:  

Others:

</code></pre>

### PR DESCRIPTION
**Pull request by griest024**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/graycoreio/daffodil/blob/develop/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Fixes: #2903

## What is the new behavior?
- moves the example viewer components to a `docs-components` package
- changes the example rendering from a custom element based one to a component outlet one
- adds custom element support for the example viewer container so it can be rendered when the markdown is rendered with `innerHtml`
- converts example components to standalone so that they can be dynamically rendered
- extracts internal package ignore constant in dgeni and adds new packages

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

## Other information

### CODE REVIEW

**Comment by griest024** [comment]
@xelaint Can you acceptance test that design-land and daffio look correct to you?

File: libs/design/paginator/examples/src/public_api.ts [hunk 1]
```diff
@@ -1,5 +1,4 @@
 export { BasicPaginatorComponent } from './basic-paginator/basic-paginator.component';
 export { LinkPaginatorComponent } from './link-paginator/link-paginator.component';
 
-export { PaginatorExamplesModule } from './paginator-examples.module';
-export { PAGINATOR_EXAMPLES } from './examples';
+export * from './examples';
```
Comment (xelaint) [inline libs/design/paginator/examples/src/public_api.ts:1]: seems like this file isn't set up like all the others and is causing errors? i've built everything and keep getting errors about paginator not being found.

```
./libs/design/paginator/examples/src/basic-paginator/basic-paginator.component.ts:10:0-65 - Error: Module not found: Error: Package path ./paginator is not exported from package /Users/elain/Desktop/Graycore/Daffodil/daffodil/libs/design (see exports field in /Users/elain/Desktop/Graycore/Daffodil/daffodil/libs/design/package.json)

./libs/design/paginator/examples/src/link-paginator/link-paginator.component.ts:16:0-65 - Error: Module not found: Error: Package path ./paginator is not exported from package /Users/elain/Desktop/Graycore/Daffodil/daffodil/libs/design (see exports field in /Users/elain/Desktop/Graycore/Daffodil/daffodil/libs/design/package.json)

Error: libs/design/paginator/examples/src/basic-paginator/basic-paginator.component.ts:6:37 - error TS2307: Cannot find module '@daffodil/design/paginator' or its corresponding type declarations.

6 import { DaffPaginatorModule } from '@daffodil/design/paginator';
                                      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~


Error: libs/design/paginator/examples/src/link-paginator/link-paginator.component.ts:12:37 - error TS2307: Cannot find module '@daffodil/design/paginator' or its corresponding type declarations.

12 import { DaffPaginatorModule } from '@daffodil/design/paginator';
                                       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

**Review by xelaint — CHANGES_REQUESTED** [review]

File: libs/design/paginator/examples/src/public_api.ts (hunk 1 above)
Comment (griest024) [reply libs/design/paginator/examples/src/public_api.ts:1]: hm odd, that error doesn't throw on my machine, even when rimrafing `dist`. Can you try again with the pushed commit?

Comment (griest024) [reply libs/design/paginator/examples/src/public_api.ts:1]: paginator was missing an `index.ts`. I added it and `daffio-dev` served correctly

File: libs/docs-components/src/code-preview/component/code-preview.component.html [hunk 2]
```diff
@@ -0,0 +1,9 @@
+@if (example) {
+  <div class="daff-docs-code-preview__content" *ngComponentOutlet="example.component"></div>
```
Comment (xelaint) [inline libs/docs-components/src/code-preview/component/code-preview.component.html]: `<div class="daff-docs-code-preview__content">` this div does not show up in the DOM.

Screen Shot 2024-07-15 at 8 51 14 PM

**Review by xelaint — CHANGES_REQUESTED** [review]

File: apps/design-land/src/scss/component-themes.scss [hunk 3]
```diff
@@ -1,4 +1,5 @@
-@use '../app/core/code-preview/component/code-preview-theme' as code-preview;
+// stylelint-disable-next-line max-line-length
+@use '@daffodil/docs-components/src/code-preview/component/code-preview-theme.scss' as code-preview;
```
Comment (xelaint) [inline apps/design-land/src/scss/component-themes.scss]: This doesn't work. I think `docs-components` package needs its own `scss` folder that has its own `component-theme.scss` and `styles.scss` files. is that right, @damienwebdev?

Also in general, I don't think these reference paths should include `.scss`?

**Review by xelaint — CHANGES_REQUESTED** [review]

File: libs/docs-components/src/code-preview/component/code-preview.component.html (hunk 2 above)
Comment (griest024) [reply libs/docs-components/src/code-preview/component/code-preview.component.html]: fixed

File: apps/design-land/src/scss/component-themes.scss (hunk 3 above)
Comment (griest024) [reply apps/design-land/src/scss/component-themes.scss]: yeah I don't know what I'm doing for this. I changed some stuff that I think is closer to the solution but it doesn't work yet

Comment (xelaint) [reply apps/design-land/src/scss/component-themes.scss]: fixed it

**Review by xelaint — APPROVED** [review]

**Pull request merged**